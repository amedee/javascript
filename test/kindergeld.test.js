const {berekenKindergeld} = require('../js/kindergeld');

describe('Berekening kindergeld', () => {

    it.each`
    aantalKinderen | maandloon | kindergeld
    ${1}           | ${1000}   | ${25}
    ${2}           | ${1000}   | ${50}
    ${3}           | ${1000}   | ${87.5}
    ${4}           | ${1000}   | ${125}
  `('should return $kindergeld for $aantalKinderen children and $maandloon income', ({
        aantalKinderen, maandloon, kindergeld
    }) => {
        expect(berekenKindergeld(aantalKinderen, maandloon)).toEqual(kindergeld);
    });

    it('should be verified with the previous snapshot', () => {
        expect(berekenKindergeld).toVerifyAllCombinations(
            [1, 2, 3, 4],
            [1000]
        );
    });

});
