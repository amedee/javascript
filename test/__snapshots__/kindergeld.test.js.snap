// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Berekening kindergeld should be verified with the previous snapshot 1`] = `
Object {
  "1,1000": 25,
  "2,1000": 50,
}
`;
